### PR TITLE
Require Node.js 10 and currify function

### PR DESCRIPTION
**Pull request by Richienb**
Not sure if _currify_ is the best way to describe this but I can't think of anything better.

Also removed test case that uses broken Node.js `process.binding` stuff.

Fixes: #16, Fixes: #19

### CODE REVIEW

**Comment by sindresorhus** [comment]
Node.js 12 tests are failing. I think you just need to give it a bit more threshold.

File: index.d.ts [hunk 1]
```diff
@@ -8,50 +8,64 @@ declare class AbortErrorClass extends Error {
...
+	) => ReturnType<FunctionType>) & {
 		/**
 		Abort pending executions. All unresolved promises are rejected with a `pThrottle.AbortError` error.
 		*/
 		abort(): void;
 	};
 
+	interface Options {
+		/**
+		Maximum number of calls within an `interval`.
```
Comment (sindresorhus) [inline index.d.ts:22]: Can you mention that the options are required? In the readme too.

Comment (Richienb) [reply index.d.ts:22]: What's the cleanest way to do that?

Comment (sindresorhus) [reply index.d.ts:22]: Probably as a note on line 45:

> Both the `limit` and `interval` options must be specified.

**Comment by sindresorhus** [comment]
Tests are failing on Node.js 14.

**Pull request merged**